### PR TITLE
Disable BusyBox bz2

### PR DESCRIPTION
**Pull request by stokito**
BB is built with enabled bzip2 which in fact not used by OpenWrt itself

See related forum thread https://forum.openwrt.org/t/is-bz2-compression-used/72120

### CODE REVIEW

**Comment by wulfy23** [comment]
it is used in image detection... which is a beneficial feature even if all targets using it become redundant... for a few Kb best to retain the capability...

possibly of use under the 'small_flash'... however for busybox... you might need a 'busybox_small' variant... which is possibly not such a silly idea... improbable for this patch alone, but stack a few up and you might be in business... 

( once you check in with the layerscape, ubnt et. al. and verify which devices are actually using bz on-device and check they are not 'small_flash' )

**Comment by stokito** [comment]
Thank you for a quick response. What exactly you mean “used for image detection”? I made a search of “DEPENDS.*bz2” on packages and nothing special was found.
As far I understood it used only in build time.

**Comment by wulfy23** [comment]
```
fgrep -r bz package/base-files/ target/linux/ | grep -v '\.patch'
```

**Comment by stokito** [comment]
Thank you for pointing out. I checked this scripts and as far I understood:
* `target/linux/layerscape/image/mkits-multiple-config.sh`  looks like used on build time
* `package/base-files/files/lib/upgrade/stage2`  directly not use but mentions `bzcat`
* `package/base-files/files/lib/upgrade/common.sh` the only one place where `bzcat` is called directly to unpack image. But from what I understood we can't compile bzipped images.

Also I found a good talk [Fight for the bytes! Fun with Four Megabytes Flash](https://youtu.be/M3yUjHKvde8?t=846) where mt mentioned that bz2 can be disabled

**Comment by adschm** [comment]
Note that apart from the discussion about the content, this has several formal issues. Please visit https://openwrt.org/submitting-patches again...

**Comment by stokito** [comment]
@adschm I changed commit message and signed. Please check if it's ok.

File: package/utils/busybox/Config-defaults.in [hunk 1]
```diff
@@ -2268,6 +2268,18 @@ config BUSYBOX_DEFAULT_FEATURE_HTTPD_PROXY
...
+	default n
+config BUSYBOX_DEFAULT_FEATURE_HTTPD_DATE
+	bool
+	default n
+config BUSYBOX_DEFAULT_FEATURE_HTTPD_LAST_MODIFIED
+	bool
+	default n
+config BUSYBOX_DEFAULT_FEATURE_HTTPD_ETAG
+	bool
+	default y
```
Comment (wulfy23) [inline package/utils/busybox/Config-defaults.in:2282]: i think you have pushed something inadvertently? these do not look related to bzcar to me...

Comment (stokito) [reply package/utils/busybox/Config-defaults.in:2282]: Omg, yes, these changes was in another branch. I’ll fix in a hour. Sorry

**Review by neheb — APPROVED** [review]

**Comment by fanck0605** [comment]
@stokito It seems that bzip is used here [/lib/upgrade/common.sh#L99](https://github.com/openwrt/openwrt/blob/331892f85f1cfab21b26a1f84824f4110e06027f/package/base-files/files/lib/upgrade/common.sh#L99)

**Comment by stokito** [comment]
> package/base-files/files/lib/upgrade/common.sh the only one place where bzcat is called directly to unpack image. But from what I understood we can't even compile bzipped images.

**Comment by CodeFetch** [comment]
I also see no point in supporting bzipped images as they are not used by OpenWrt.

@stokito So maybe add a commit for removing support of bzipped images...

BTW https://github.com/openwrt/openwrt/pull/3940

**Comment by stokito** [comment]
@CodeFetch thank you for the comment. I just not experienced in that area so I can't add a such commit. If you do feel it is needed then please send a PR for this.

**Comment by CodeFetch** [comment]
@stokito If you want to disable bz2 support, you also need to remove the part which depends on it:
https://github.com/openwrt/openwrt/blob/331892f85f1cfab21b26a1f84824f4110e06027f/package/base-files/files/lib/upgrade/common.sh#L99

Otherwise your PR is leaving dead code and won't be merged.

**Comment by stokito** [comment]
@CodeFetch  as you proposed I removed the support of bzipped images and force pushed the resulted commit 6ed6152
Please review

**Comment by wulfy23** [comment]
is this artifact intentional?
```
CONFIG_BUSYBOX_DEFAULT_BZIP2_SMALL=0
```
please provide example of claimed 8K saving

**Comment by stokito** [comment]
@wulfy23 in fact I disabled not `CONFIG_BUSYBOX_DEFAULT_BZIP2_SMALL=0` but `BUSYBOX_DEFAULT_FEATURE_BZIP2_DECOMPRESS` and yes, I did it by mistake. Thank you for noticing!

> please provide example of claimed 8kb saving
busybox menuconfig
I didn't checked it myself because openwrt doesn't have `make bloatcheck` as BusyBox have.
This will also safe a build time a little bit.

**Comment by aparcar** [comment]
Merged, thank you!

**Comment by stokito** [comment]
Thank you all! Now we must not to forget to add notice to change log
